<a href="https://colab.research.google.com/github/Manuel-Navarro-Barrera/Probabilidad_y_Estadistica/blob/main/Unidad%204/Problemas_de_Distribuci%C3%B3n_de_Poisson_y_Binomial.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Problemas de Distribución de Poisson y Binomial**

---








# **Problema 1**
#### Antes de resolver el problema especificar si la solucion es de tipo Binomial o Poisson y explicar por que.
##### Realizar un programa en Python que resuelva el sigueinte problema:
*   El número x de personas ingresadas a una unidad de cuidados intensivos en un hospital particular, en un dia, tiene una distribucion de probabilidad de Poisson con media igual a 5 personas por dia.
*   ¿Cuál es la probabilidad de que el número de personas ingresadas a una unidad de cuidados intensivos en un hospital particular, en un dia particular, osea 2?
*   ¿Es probable que x exceda de 10? Explicar

## **Explicacion si es Binomial o Poisson**

#### -Es de Poisson ya que estamos contando el número de veces que ocurre un evento (ingresos a cuidados intensivos) en un intervalo de tiempo fijo (un día).
#### -También se conoce una tasa media de ocurrencia (λ=5).
#### -Además los eventos son independientes y raros, y no existe un número maximo de intentos.



In [1]:
import math
from scipy.stats import poisson

In [2]:
# Parámetro de Poisson
lambda_poisson = 5

# a) Probabilidad de que ingresen exactamente 2 personas
k = 2
probabilidad_a = poisson.pmf(k, mu=lambda_poisson)

print(f"a) La probabilidad de que ingresen exactamente 2 personas es: {probabilidad_a:.4f}")

a) La probabilidad de que ingresen exactamente 2 personas es: 0.0842


In [3]:
# b) Probabilidad de que X > 10
probabilidad_b = 1 - poisson.cdf(10, mu=lambda_poisson)
print(f"b) La probabilidad de que ingresen más de 10 personas es: {probabilidad_b:.4f}")

if probabilidad_b > 0.05:
    print("-Es algo probable que se exceda de 10 ingresos en un día.")
else:
    print("-No es muy probable que se exceda de 10 ingresos en un día.")

b) La probabilidad de que ingresen más de 10 personas es: 0.0137
-No es muy probable que se exceda de 10 ingresos en un día.


# **Pregunta a)**

Usamos la fórmula de Poisson:

𝑃
(
𝑋
=
2
)
=
𝑒
^{-5}
⋅
5
^{-2}/
2
!
≈
0.0842

##### Por lo tanto la respuesta es: La probabilidad de que ingresen exactamente 2 personas en un día es aproximadamente 0.0842 (8.42%).

---



# **Pregunta b)**

#### Calculamos: P(X>10)=1−P(X≤10):

## P(X > 10) ≈ 1 − 0.9863 = 0.0137
##### La respuesta es: La probabilidad de que ingresen más de 10 personas en un día es aproximadamente 0.0137 (1.37%), lo cual es muy poco probable.

---

# **Problema 2**

#### Antes de resolver el problema especificar si la solucion es de tipo Binomial o Poisson y explicar por que.
##### Realizar un programa en Python que resuelva el sigueinte problema:

*   Un fabricante de podadoras para el pasto compra motores de 1 hp y 2 ciclos, en lotes de 1000, a un proveedor.
*   El fabricante equipa cada una de las podadoras producidas por su planta con uno de los motores.
*   La historia muestra que la probabilidad de que cualquier motor del proveedor resulte no satisfactorio es .001.
*   **En un embarque de 1000 motores, Calcular:**
*   ¿Cuál es la probabilidad de que ninguno sea defectuoso?,
*   ¿Cuál es la probabilidad de que tres sean defectuosos?,
*   ¿Cuál es la probabilidad de que cuatro sean defectuosos?













## **Explicacion si es Binomial o Poisson**

#### -El problema original es Binomial, porque hay un número fijo de intentos (1000 motores), donde cada motor tiene dos posibles resultados: defectuoso o no defectuoso. Su probabilidad de defecto es constante: 𝑝=0.001 y los eventos son independientes.
#### -Sin embargo, como **n** es grande (1000) y **p** es muy pequeño (0.001), se puede aproximar la Binomial con una distribución de Poisson:
#### Si 𝑛 es grande y 𝑝 es pequeño, con λ=np, entonces:
𝐵𝑖𝑛𝑜𝑚𝑖𝑎𝑙(𝑛,𝑝)≈𝑃𝑜𝑖𝑠𝑠𝑜𝑛(𝜆=𝑛𝑝)
#### -En este caso, usamos Poisson con λ=1000*0.001=1



In [11]:
lambda_poisson = 1

In [12]:
prob_0_defectuosos = poisson.pmf(0, mu=lambda_poisson)
print(f"-Probabilidad de que ningún motor sea defectuoso: {prob_0_defectuosos:.4f}")

prob_3_defectuosos = poisson.pmf(3, mu=lambda_poisson)
print(f"-Probabilidad de que tres motores sean defectuosos: {prob_3_defectuosos:.4f}")

prob_4_defectuosos = poisson.pmf(4, mu=lambda_poisson)
print(f"-Probabilidad de que cuatro motores sean defectuosos: {prob_4_defectuosos:.4f}")

-Probabilidad de que ningún motor sea defectuoso: 0.3679
-Probabilidad de que tres motores sean defectuosos: 0.0613
-Probabilidad de que cuatro motores sean defectuosos: 0.0153


# **Pregunta 1**
### *¿Cuál es la probabilidad de que tres motores sean defectuosos?*

###Calculamos 𝑃 ( 𝑋 = 0 ) :
𝑃(𝑋=0)=𝑒−1⋅1^{0}/0!=𝑒−1≈0.3679

#### Por lo tanto, la probabilidad de que ningún motor sea defectuoso es aproximadamente 0.3679 (36.79%).

---



# **Pregunta 2**

### *¿Cuál es la probabilidad de que tres motores sean defectuosos?*

P(X=3)= e^{−1}⋅1^{3} / 3! = e^{−1}/ 6 ​≈0.0613

#### Dicho este calculo, la probabilidad de que tres motores sean defectuosos es aproximadamente 0.0613 (6.13%).

---




# **Pregunta 3**

### *¿Cuál es la probabilidad de que cuatro motores sean defectuosos?*

P(X=4) = e ^{−1}*1 ^{4} / 4! ​= e ^{−1} / 24​ ≈ 0.0153

#### Dicho de esta formula la probabilidad de que cuatro motores sean defectuosos es aproximadamente 0.0153 (1.53%).